python-sdk/src/astro/databricks/delta.py

```diff
@@ -100,6 +102,17 @@ def create_schema_if_needed(self, schema: str | None) -> None:
         # Schemas do not need to be created for delta, so we don't need to do anything here
         return None
 
+    def fetch_all_rows(self, table: BaseTable) -> list:
```

Comment:
Same comment as in the base table - should we have an upper bound here?

Reply:
I don't think having an upper bound is a good idea as it will create false information for users, but I can see having upper_bound as an option. As to the risk of users loading too much data, I think the function is pretty self explanatory and they should consider the risks (especially since the Database object isn't really exposed to users unless they look for it).